src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -397,28 +388,17 @@ public static void CreateAssociation(object primary, object secondary)
                 throw new ArgumentException(SR.TypeDescriptorSameAssociation);
             }
 
-            if (s_associationTable == null)
-            {
-                lock (s_internalSyncObject)
-                {
-                    if (s_associationTable == null)
-                    {
-                        s_associationTable = new WeakHashtable();
-                    }
-                }
-            }
-
-            IList associations = (IList)s_associationTable[primary];
+            IList associations = (IList)s_associationTableLazy.Value[primary];
 
             if (associations == null)
             {
-                lock (s_associationTable)
+                lock (s_associationTableLazy)
```

Comment:
On the lines of `s_defaultProviders` reversal discussion above, does `s_associationTable` really require lazy initialisation. The initialisation is just object creation and does not do any heavy code that may cause delays. So how about making it readonly and initialize in-place?

Extending a bit further, do we want to say that lazy initialisation using `Lazy<T>` or `LazyInitializer.EnsureInitialized` makes more sense if the initialisation is non-trivial. (E.g. s_ProviderTable and s_ProvideTypeTable) in this same file which I left untouched. s_associationTable could also be treated in same way....?